discovery/kubernetes/endpoints.go

```diff
@@ -47,11 +48,11 @@ type Endpoints struct {
 	endpointsStore cache.Store
 	serviceStore   cache.Store
 
-	queue *workqueue.Type
+	queue *workqueue.Typed[any]
 }
 
 // NewEndpoints returns a new endpoints discovery.
-// Endpoints API is deprecated in k8s v1.33+, but we should still support it.
+// Deprecated: The Endpoints API is deprecated starting in K8s v1.33+. Use NewEndpointSlice.
```

Comment:
I've added a deprecation notice and nudge to use the `EndpointSlice` objects instead based on @machine424 's feedback in https://github.com/prometheus/prometheus/pull/16890#issuecomment-3302508487